portal/common/lib/routing.ts

```diff
@@ -5,115 +5,123 @@ import { SuiObjectResponse } from "@mysten/sui/client";
+     *
+     * @param siteObjectId - The ID of the site object.
+     * @returns The routes list.
+     */
+    public async getRoutes(
+        siteObjectId: string,
+    ): Promise<Routes | undefined> {
+        logger.info({ message: "Fetching routes dynamic field.", siteObjectId })
+        const routesDF = await this.fetchRoutesDynamicField(siteObjectId);
+        if (!routesDF.data) {
+            logger.warn({
+                message: "No routes dynamic field found for site object. Exiting getRoutes.",
+                siteObjectId
+            });
+            return;
+        }
+        const routesObj = await this.fetchRoutesObject(routesDF.data.objectId);
+        const objectData = routesObj.data;
+        if (objectData && objectData.bcs && objectData.bcs.dataType === "moveObject") {
+            return this.parseRoutesData(objectData.bcs.bcsBytes);
+        }
+        if (!objectData) {
+            logger.warn({
+                message: "Routes dynamic field does not contain a `data` field."
+            });
+        } else if (!objectData.bcs) {
+            logger.warn({
+                message: "Routes dynamic field does not contain a `bcs` field."
+            });
+        } else if (!objectData.bcs.dataType) {
+            logger.warn({
+                message: "Routes dynamic field does not contain a `dataType` field."
+            });
+        }
+        throw new Error("Routes object data could not be fetched.");
     }
-    if (!objectData) {
-        logger.warn({
-            message: "Routes dynamic field does not contain a `data` field."
-        });
-    } else if (!objectData.bcs) {
-        logger.warn({
-            message: "Routes dynamic field does not contain a `bcs` field."
+
+    /**
+     * Fetches the dynamic field object for routes.
+     *
+     * @param client - The SuiClient instance.
+     * @param siteObjectId - The ID of the site object.
+     * @returns The dynamic field object for routes.
+     */
+    private async fetchRoutesDynamicField(
+        siteObjectId: string,
+    ): Promise<SuiObjectResponse> {
+        return await this.rpcSelector.getDynamicFieldObject({
+            parentId: siteObjectId,
+            name: { type: "vector<u8>", value: "routes" },
         });
-    } else if (!objectData.bcs.dataType) {
-        logger.warn({
-            message: "Routes dynamic field does not contain a `dataType` field."
+    }
+
+    /**
+     * Fetches the routes object using the dynamic field object ID.
+     *
+     * @param client - The SuiClient instance.
+     * @param objectId - The ID of the dynamic field object.
+     * @returns The routes object.
+     */
+    private async fetchRoutesObject(objectId: string): Promise<SuiObjectResponse> {
+        return await this.rpcSelector.getObject({
+            id: objectId,
+            options: { showBcs: true },
         });
     }
-    throw new Error("Routes object data could not be fetched.");
-}
```

Comment:
Question: since we know the name of the dynamic field, why do we do two full node queries here, instead of computing the address of the dynamic field directly?

Reply:
Good catch! I am not sure why we missed this one in the past. I will look into it and replace it with `deriveDynamicField`. 🤔

Reply:
I will add this in a later PR: https://github.com/MystenLabs/walrus-sites/pull/332

I did not made the commit to this branch because I was not able to test that the dynamic fields work due to the full node bug.